src/dmd/hdrgen.d

```diff
@@ -250,6 +250,20 @@ public:
     override void visit(WhileStatement s)
     {
         buf.writestring("while (");
+        if (auto p = s.param)
+        {
+            // Print condition assignment
+            StorageClass stc = p.storageClass;
+            if (!p.type && !stc)
+                stc = STC.auto_;
+            if (stcToBuffer(buf, stc))
+                buf.writeByte(' ');
+            if (p.type)
+                typeToBuffer(p.type, p.ident, buf, hgs);
+            else
+                buf.writestring(p.ident.toString());
```

Comment:
This else looks wrong. It should print the name unconditionally, and add an extra parenthesis after `typeToBuffer`. Try `while (scope int x = foo())` to see the bug.

Reply:
It's not wrong. You can have: `while (int a = fun())` in which case the `if` is taken and it prints the type and the identifier (as it is passed as a parameter and typeToBuffer prints it if is non-null). Or you can have `while (const a = fun())` in which case the else branch is taken. This the correct behavior and it is exactly the same as the `if` one.